### PR TITLE
Avoid strlen for admin submenu items

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #43715, #39758

Credits @Hackwar

### Summary of Changes
Remove empty values from request  

### Testing Instructions
see #43715, #39758

### Actual result BEFORE applying this Pull Request
Error message when adding or editing a menuItemType in some cases - see description in #43715, #39758

### Expected result AFTER applying this Pull Request
MenuItemTypes can be selected and edited

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Review by EJBJane — COMMENTED** [review]
I tried the code change but it gives this. So maybe its not right? 

[MenutypesModel.txt](https://github.com/user-attachments/files/16507918/MenutypesModel.txt)

screencapture-hdm-agency-administrator-index-php-2024-08-06-10_48_33

File: administrator/components/com_menus/src/Model/MenutypesModel.php [hunk 1]
```diff
@@ -407,9 +407,12 @@ protected function getTypeOptionsFromManifest($component)
                 $request['view']       = (string) $attributes->view;
                 $request['layout']     = (string) $attributes->layout;
                 $request['sub']        = (string) $attributes->sub;
+
+                // Remove empty values.
+                $request = array_filter($request);
             }
 
-            $o->request = array_filter($request, 'strlen');
+            $o->request = $request;
```
Comment (Fedik) [inline administrator/components/com_menus/src/Model/MenutypesModel.php]: I think we still need to filter all values.
Please change to following:
```php
$o->request = array_filter($request, function($value) {
    if (is_array($value)) {
        return !empty($value);
    } else {
        return strlen($value);
   }
});
```

Comment (EJBJane) [reply administrator/components/com_menus/src/Model/MenutypesModel.php]: It works for me now..

Comment (chmst) [reply administrator/components/com_menus/src/Model/MenutypesModel.php]: thank you

File: administrator/components/com_menus/src/Model/MenutypesModel.php [hunk 2]
```diff
@@ -409,7 +409,14 @@ protected function getTypeOptionsFromManifest($component)
                 $request['sub']        = (string) $attributes->sub;
             }
 
-            $o->request = array_filter($request, 'strlen');
+            $o->request = array_filter($request, function($value) {
+                if (is_array($value)) {
+                    return !empty($value);
+                } else {
+                    return strlen($value);
```
Comment (Quy) [inline administrator/components/com_menus/src/Model/MenutypesModel.php]:
```suggestion
                }

                return strlen($value);
```

File: administrator/components/com_menus/src/Model/MenutypesModel.php [hunk 3]
```diff
@@ -409,7 +409,15 @@ protected function getTypeOptionsFromManifest($component)
...
             }
 
-            $o->request = array_filter($request, 'strlen');
+            $o->request = array_filter($request, function($value) {
+                if (is_array($value)) {
+                    return !empty($value);
+                }
+
+                return strlen($value);
+               }
```
Comment (Quy) [inline administrator/components/com_menus/src/Model/MenutypesModel.php]:
```suggestion
```

File: administrator/components/com_menus/src/Model/MenutypesModel.php [hunk 4]
```diff
@@ -409,7 +409,14 @@ protected function getTypeOptionsFromManifest($component)
                 $request['sub']        = (string) $attributes->sub;
             }
 
-            $o->request = array_filter($request, 'strlen');
+            $o->request = array_filter($request, function($value) {
```
Comment (Quy) [inline administrator/components/com_menus/src/Model/MenutypesModel.php]:
```suggestion
            $o->request = array_filter($request, function ($value) {
```

**Comment by Quy** [comment]
Please rebase for 5.2.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on df46b340ef4d62be780571c15faab76db994b9af<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43884">issues.joomla.org/tracker/joomla-cms/43884</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on df46b340ef4d62be780571c15faab76db994b9af<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43884">issues.joomla.org/tracker/joomla-cms/43884</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43884">issues.joomla.org/tracker/joomla-cms/43884</a>.</sub>

**Pull request merged**